contracts/MetaIdentityManager.sol

```diff
@@ -0,0 +1,211 @@
+    if (recoveryKeys[identity] == sender) _;
+    else throw;
+  }
+
+  modifier rateLimited(Proxy identity, address sender) {
+    if ((limiter[identity][sender] + adminRate) < now) {
+      limiter[identity][sender] = now;
+      _ ;
+    } else throw;
+  }
+
+  modifier validRecovery(address newKey) { //protects against some weird attacks
+    if (newKey != address(0)) _;
+    else throw;
+  }
+
+  /// @dev Contract constructor sets initial timelocks and meta-tx relay address
+  /// @param _userTimeLock Time before new owner can control proxy
+  /// @param _adminTimeLock Time before new owner can add/remove owners
+  /// @param _adminRate Time period used for rate limiting a given key for admin functionality
+  /// @param _relayAddress Address of meta transaction relay contract
+  function MetaIdentityManager(uint _userTimeLock, uint _adminTimeLock, uint _adminRate, address _relayAddress) {
+    adminTimeLock = _adminTimeLock;
+    userTimeLock = _userTimeLock;
+    adminRate = _adminRate;
+    relay = _relayAddress;
+  }
+
+  /// @dev Creates a new proxy contract for an owner and recovery
+  /// @param owner Key who can use this contract to control proxy. Given full power
+  /// @param recoveryKey Key of recovery network or address from seed to recovery proxy
+  /// Gas cost of ~300,000
+  function CreateIdentity(address owner, address recoveryKey) validRecovery(recoveryKey) {
+    Proxy identity = new Proxy();
+    owners[identity][owner] = now - adminTimeLock; // This is to ensure original owner has full power from day one
+    recoveryKeys[identity] = recoveryKey;
+    IdentityCreated(identity, msg.sender, owner,  recoveryKey);
+  }
+
+  /// @dev Allows a user to transfer control of existing proxy to this contract. Must come through proxy
+  /// @param owner Key who can use this contract to control proxy. Given full power
+  /// @param recoveryKey Key of recovery network or address from seed to recovery proxy
+  /// Note: User must change owner of proxy to this contract after calling this
+  function registerIdentity(address owner, address recoveryKey) validRecovery(recoveryKey) {
+    if (recoveryKeys[msg.sender] > 0 ) throw; // Invariant enforced w/ validRecovery modifier
+    owners[msg.sender][owner] = now - adminTimeLock; // Owner has full power from day one
+    recoveryKeys[msg.sender] = recoveryKey;
+    IdentityCreated(msg.sender, msg.sender, owner, recoveryKey);
+  }
+
+  /// @dev Allows a user to forward a call through their proxy.
+  function forwardTo(address sender, Proxy identity, address destination, uint value, bytes data)
+    onlyAuthorized
+    onlyOwner(identity, sender)
+  {
+    identity.forward(destination, value, data);
+  }
+
+  /// @dev Allows an olderOwner to add a new owner instantly
+  function addOwner(address sender, Proxy identity, address newOwner)
+    onlyAuthorized
+    onlyOlderOwner(identity, sender)
+    rateLimited(identity, sender)
+  {
+    owners[identity][newOwner] = now;
```

Comment:
Good point - I missed this in my comment. I don't see a problem w/ adding new owner instantly. What do you think @oed ?